### ISSUE TITLE
Add accent insensitive query mode

### ISSUE THREAD

**Issue by friebetill** [prisma/prisma#6428]
## Problem

Currently there is a possibility to search a database case insensitive with `QueryMode.insensitive`. I would like to additionally search the database accent insensitive.

## Suggested solution

Extend the `QueryMode.insensitive` to be also accent insensitive or add the `QueryMode.accentInsensitive` with the possibility to combine the mode with `QueryMode.insensitive` (it would probably be good to rename `QueryMode.insensitive` to `QueryMode.caseInsensitive`)

**Comment by matthewmueller** [comment]
Hey @friebetill, I can understand the use case. This is unfortunately a very complex subject and our current implementation is best-effort and depends on a lot of factors. 

Do you know how you would do this in SQL? If we have the SQL and potentially database collation settings, we can have a closer look.

**Comment by friebetill** [comment]
Thanks for your answer! So far I haven't implemented anything, but I will probably follow [this Stack Overflow answer](https://stackoverflow.com/a/11007216) and use the extension `unaccent` to solve it. I hope that this extension can be installed directly in Postgres and then used with the Prisma client via `$queryRaw`.

Of course, the approach does not work for the other databases and the first search result, e.g. [for MySQL](https://stackoverflow.com/questions/41086590/postgresql-unaccent-equivalent-in-mysql) is not promising.

**Comment by theduardomaciel** [comment]
@friebetill If you've had any success with installing the module and using the function in a $rawQuery (and still have access to it, because it's been a while since you posted), please share here because I'm racking my brains to find out how to implement this :)

**Comment by friebetill** [comment]
Unfortunately, I didn't try to solve the problem further because my capacity is limited.

**Comment by sostenesapollo** [comment]
Any updates ?

**Comment by Ben212** [comment]
It will be very helpful to get "unaccent" from postgreSQL working without having to use query raw

**Comment by kefniark** [comment]
Yes same thing, the whole search and fulltextsearch features of prisma become kinda useless at the moment you have non ASCII characters contents.
When you have multi language contents (japanese, chinese, ...) It become impossible to do a decent db search with default prisma tools and casing.

It's frustrating to rely on raw query just to wrap the field you search or filter `lower(unaccent(title))`.

**Comment by andrefelipeschulle** [comment]
Any news on this Unaccent use case?

**Comment by Cauen** [comment]
To be able to use "unnacent" with a complex/typesafe "where", I'm using 2 queries: 

1. The first using "queryRawUnsafe" that returning IDs
2. The second using these IDs to compose a native Prisma query.

Hope it helps someone :)
Does anyone have a better approach?

[Example](https://gist.github.com/Cauen/72a03ecbbe5b06ab70bb3b24fe6648ab)

**Comment by luizfelipelaviola** [comment]
Accent insensitive search is very important for languages like Portuguese and others.

When trying to search "João" people may use "joao" and this should be handled by prisma to avoid doing a lot of raw queries to make that work!

**Comment by lmalvasia** [comment]
Any news about this?

**Comment by gandrin** [comment]
> Hey @friebetill, I can understand the use case. This is unfortunately a very complex subject and our current implementation is best-effort and depends on a lot of factors.
> 
> Do you know how you would do this in SQL? If we have the SQL and potentially database collation settings, we can have a closer look.

Hello there, I'm also in need of searching without having to think about accent.

Unfortunatly, I didn't have the chance yet to deep dive into the code of Prisma. So maybe what I am about to say is retarded :)

If it's complicated to handle it for all languages, maybe we could settle for a solution where Prisma provides a hook called when the unaccent parameter is set? The objective of this function would be to locally modify the query and apply PSQL functions like unaccent ?

It would prevent the use of raw query for the time being, and once the feature is fully operational, the hook can be ignored.

**Comment by vanflux** [comment]
Ahhhh, remembering how I love modern ORMs...
"Prisma is a next-generation Node.js and TypeScript ORM that unlocks a **new level of developer experience**"

**Comment by itihell** [comment]
with queryraw you can do it like this

const rows = await prisma.$queryRaw`
      select *
      from personas 
      where translate(nombres,'áéíóúÁÉÍÓÚäëïöüÄËÏÖÜ','aeiouAEIOUaeiouAEIOU') ILIKE '%' || translate(${search}, 'áéíóúÁÉÍÓÚäëïöüÄËÏÖÜ','aeiouAEIOUaeiouAEIOU') || '%' 
      or translate(apellidos,'áéíóúÁÉÍÓÚäëïöüÄËÏÖÜ','aeiouAEIOUaeiouAEIOU') ILIKE '%' || translate(${search}, 'áéíóúÁÉÍÓÚäëïöüÄËÏÖÜ','aeiouAEIOUaeiouAEIOU') || '%' limit 30`;

**Comment by victorxavier10** [comment]
Any news? Let's speed up this issue!!

**Comment by jdvalenzuelap** [comment]
Hey! I think this is a must for latin languages. A proper search can't be done without this.

**Comment by agustin-sanc** [comment]
A solution is to save two props, one with accents and another without accents. Use the prop without accents for the search, and the prop with accents to display in the UI.

**Comment by sostenesapollo** [comment]
@agustin-sanc that's not the solution, workarround, but for a large database it doesn't work, data duplicity

**Comment by pablojsx** [comment]
Still waiting : )

**Comment by anfelipeloal** [comment]
Any update? As many have already said, this will be a really useful for multilingual applications, where store data from people in different countries and use accented letters

**Comment by Cauen** [comment]
Hi guys, while waiting, i've figured and been using a good alternative using [Views](https://www.prisma.io/docs/orm/prisma-schema/data-model/views)
I'm using since **October 24th, 2023** (Since the View feature was still in preview, but now its Generally Available) and to this day with 0 problems

It:
✅ Keeps Prisma type safety
✅ Do only 1 query to database

The idea is:

1. Create a linked-to-model Prisma View
```prisma
// schema.prisma
view PersonView {
  id                 String  @id @unique
  normalizedName     String?  
  normalizedNickname String?
  Person             Person  @relation(fields: [id], references: [id])
}
```
2. Create the view with "lower_unnacent" texts
```sql
-- migration.sql
CREATE EXTENSION unaccent;

CREATE
  OR REPLACE FUNCTION public.lower_unaccent (text) RETURNS text LANGUAGE plpgsql IMMUTABLE AS $function$
  BEGIN
      RETURN lower(unaccent($1));
  END;
  $function$;

CREATE VIEW "PersonView" AS
    SELECT
      id,
      lower_unaccent(name) AS "normalizedName",
      lower_unaccent(nickname) AS "normalizedNickname"
    FROM
      "Person";
```
3. When filtering, get data from the view
```ts
// common/utils/string.ts
export const unnacentLower = (str: string) =>
  str
    .normalize("NFD")
    .replace(/[\p{Diacritic}]/gu, "")
    .toLocaleLowerCase()

// query.ts
// ...
person: {
  PersonView: {
    OR: [
      {
        normalizedName: {
          contains: unnacentLower(textSearch),
        },
      },
      {
        normalizedNickname: {
          contains: unnacentLower(textSearch),
        },
      },
    ],
  },
},
```

Hope it can help someone :)

**Comment by pearl-nguyen** [comment]
Any update on this issue? My team uses MS SQL Server and we need prisma support for accent insensitive filtering to be used in multilingual online applications.

**Comment by Batocanin** [comment]
4.5 years later and it was still not supported, eh..?

**Comment by tsito-irok** [comment]
2026, yet not supported, uh !?

**Comment by danesto** [comment]
Any way we can help develop and speed this up? ⏩

**Comment by vanflux** [comment]
> Any way we can help develop and speed this up? ⏩

Just create an auto generated stored column and apply unnacent and any normalization you want. Then filter based on that column.

**Comment by danesto** [comment]
I know, but I'd have to rewrite the whole search strategy for that, on the app thats on production already and potentially migrate existing data to the autogenerated column.
I might just end up refactoring to use unnacent with raw query for now.

**Comment by brucePedroGomes** [comment]
Hey everyone, hope you're all doing well! We're already halfway through 2026 and this still hasn't been resolved. I'd genuinely like to know if there's actually a plan in place to address this

**Comment by icausera** [comment]
Would be very helpful for Spanish too — á é í ó ú ü ñ accent-insensitive matching is a common requirement, and the workarounds add real complexity and an extra round trip just to search a text field. Is there a roadmap for native unaccent-based accent-insensitive filtering (e.g. an accentInsensitive mode alongside insensitive) so this doesn't require workarounds?
